src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs

```diff
@@ -1058,7 +1052,7 @@ internal NamedTypeSymbol Construct(ImmutableArray<TypeWithAnnotations> typeArgum
 
             Debug.Assert(!unbound || typeArguments.All(TypeWithAnnotationsIsErrorType));
 
-            if (ConstructedNamedTypeSymbol.TypeParametersMatchTypeArguments(this.TypeParameters, typeArguments))
+            if (!constructWithTypeParameters && ConstructedNamedTypeSymbol.TypeParametersMatchTypeArguments(this.TypeParameters, typeArguments))
```

Comment:
>constructWithTypeParameters [](start = 17, length = 27)

Is `constructWithTypeParameters` needed? It looks like even if `constructWithTypeParameters` is set, we still construct with the given `typeArguments`.